lib/compiler/build_runner.zig

```diff
@@ -202,6 +202,15 @@ pub fn main() !void {
                         next_arg, @errorName(err),
                     });
                 };
+            } else if (mem.eql(u8, arg, "--build-id")) {
+                builder.build_id = .fast;
```

Comment:
Just out of curiosity, why this default in particular? Is there precedent in other build systems / toolchains?

Reply:
Fedora (and I assume anything in the same family) explicitly uses sha1

Arch builds gcc with `--enable-linker-build-id` which automatically passes `--build-id` to the linker, implying sha1

it might be best to just remove the default here and to demand that the type is explicit

Reply:
Some context: https://github.com/ziglang/zig/pull/15459#issuecomment-1523309866

At least according to `man ld`, that comment is wrong.

I think it would be good to verify what `ld` and `ld.lld` do here (hopefully they agree), and then change Zig's default, both here and in `src/main.zig`, to be the same.

Reply:
That's fun. So in summary, defaults look as follows:

* Mach-O doesn't seem to have a build ID mechanism.
* `ld`, `lld`, and `mold` all seem to agree on `sha1` for ELF.
* `ld` uses `md5` for COFF, `lld` uses `xx` (both written as a "GUID").
* `lld` uses `fast` for WASM (UUID v5 based on xxHash).

Reply:
Okay, I've argued myself into thinking that `fast` is a good default because it's consistent (ish) across binary formats.

The only missing piece here is that `src/link/Coff.zig` needs to pass `--build-id` to LLD when the style is `fast`. (LLD does not appear to support other styles for COFF at the moment, even though all the non-`sha1` styles *could* be made to work.) Would you mind making that change?